auto_invoice_advance/models/sale_order.py

```diff
@@ -9,32 +9,99 @@
 class SaleOrder(models.Model):
     _inherit = 'sale.order'
 
+    down_payment_id = fields.Many2one('account.move',
+                                      string="Down payment invoice",
+                                      readonly=True,
+                                      help="Auto generated down payment invoice")
+
+    @api.returns('self', lambda value: value.id)
+    def copy(self, default=None):
+        # OVERRIDE
+
+        if default is None:
+            default = {}
+
+        # Do not copy down_payment_id from other sale orders.
+        if self.down_payment_id:
+            default['down_payment_id'] = None
+
+        return super(SaleOrder, self).copy(default)
+
     def _cron_create_advance_invoices(self):
         order_ids = self.search(self._prepare_order_deposit_domain())
         if len(order_ids) == 0:
             return  # Nothing to be processed
         _logger.info(f"Creating {len(order_ids)} advance invoices for sale.orders: {order_ids.ids}")
```

Comment:
Does the `len()` really add value if you see the ids printed?

Reply:
Just informative. It get's logged again at the end of the cron job. It's a nice visual confirmation in the logger that all invoices that _should_ have been created have actually been created. 

With a scenario where you have 10+ invoices, you don't have to compare the full ids list.